test/e2e/utils.go

```diff
@@ -693,3 +767,97 @@ func envoyReplicas(ctx context.Context, cl client.Client, ns string) (int32, err
 	}
 	return ds.Status.NumberReady, nil
 }
+
+func waitForIngressLB(ctx context.Context, cl client.Client, timeout time.Duration, ns, name string) (string, error) {
```

Comment:
A docstring here would be nice, I think, since the behavior is not super obvious.

In addition, when you're testing on GKE, this probably won't work so well because the hostname will be the load balancer IP. We might also need to allow supplying an override hostname for the tests or something.

Reply:
> A docstring here would be nice, I think, since the behavior is not super obvious.

ack

> when you're testing on GKE, this probably won't work so well because the hostname will be the load balancer IP.

@youngnick `podWaitForHTTPResponse()` uses "local.projectcontour.io" as the host header to support hostname or IP-based ingress e2e testing.